pkg/chartutil/create.go

```diff
@@ -52,29 +52,31 @@ const (
 const defaultValues = `# Default values for %s.
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
+
 replicaCount: 1
+
 image:
   repository: nginx
   tag: stable
   pullPolicy: IfNotPresent
+
 service:
-  name: nginx
   type: ClusterIP
-  externalPort: 80
-  internalPort: 80
+  port: 80
```

Comment:
mostly curious, what's the motivation for consolidating these two?

Reply:
In most cases, it is not necessary and probably doesn't make sense to make the container port configurable. So, I thought it's best to make things simpler here. For this example, changing the container port would break the chart because it would require changing the Nginx configuration as well.

Reply:
Yep, I agree, I actually never liked that the internalPort was configurable. Also noticed that you're not using `.Values.service.port` in the Deployment template which makes sense to me, as you would want to keep that the same (so as not to break the service) and change the k8s Service port.

Reply:
Some charts need to run internal port on a different port for security reasons like anything running under a non root user for ports less than 1024.
Removing that as a option will break a lot of containers.

Reply:
@rendhalver this is just removing the option to be configured when a user deploys a chart, the chart author should make a good decision on what internal port should use to avoid issues like non-root user support.

Reply:
@prydonius So this won't change people ability to set an InternalPort in their charts as needed?

Reply:
@rendhalver yes, and remember that this is just a sample chart. If the chart author thinks adding an internalPort value is necessary for the thing they're packaging they still can.

Reply:
Awesome. Thanks for the explanation. :)